synapse/storage/databases/main/events_bg_updates.py

```diff
@@ -1109,21 +1112,20 @@ async def _event_arbitrary_relations(
         last_event_id = progress.get("last_event_id", "")
 
         def _event_arbitrary_relations_txn(txn: LoggingTransaction) -> int:
-            # Iterate over events which do not appear in the event_relations
-            # table -- they might have a relation that was not previously stored
-            # due to an unknown relation type.
+            # Fetch events and then filter based on whether the event has a
+            # relation or not.
             txn.execute(
                 """
                 SELECT event_id, json FROM event_json
-                LEFT JOIN event_relations USING (event_id)
-                WHERE event_id > ? AND event_relations.event_id IS NULL
+                WHERE event_id > ?
                 ORDER BY event_id LIMIT ?
                 """,
                 (last_event_id, batch_size),
             )
 
             results = list(txn)
-            missing_relations = []
+            # (event_id, parent_id, rel_type) for each relation
+            missing_relations: List[Tuple[str, str, str]] = []
```

Comment:
we don't know if they are actually missing? maybe just `relations`?

Reply:
Fair. Maybe `relations_to_insert` or something. 🤷